group_utils/README.md

```diff
@@ -0,0 +1,171 @@
+var groupLabels = getMembersLabels(group)
+```
+
+### getMembersLabelsString
+This function returns the __labels__ of direct group members as a concatenated string.
+```javascript
+var groupLabelsString = getMembersLabelsString(group)
+```
+
+### getAllMembersNames
+This function returns the __names__ of direct and child group members as an array.
+The group items are excluded.
+```javascript
+var groupAllNames = getAllMembersNames(group)
+```
+
+### getAllMembersStates
+This function returns the __states__ of direct and child group members as an array.
+The group items are excluded.
+```javascript
+var groupAllStates = getAllMembersStates(group)
+```
+
+### getAllMembersLabels
+This function returns the __labels__ of direct and child group members as an array.
+The group items are excluded.
+```javascript
+var groupAllLabels = getAllMembersLabels(group)
+```
+
+### getAllMembersLabelsStrings
+This function returns the __labels__ of direct and child group members as a concatenated string.
+The group items are excluded.
+```javascript
+var groupAllLabelsString = getAllMembersLabelsString(group)
+```
+
+### __Arithmetic operations__
+Perform arithmetic operations on the states of Number members. 
+This functionality is the same as in the openHAB [group item definition](https://www.openhab.org/docs/configuration/items.html#derive-group-state-from-member-items).
+The following functions only use Items of type: Number, Dimmer or Rollershutter.
+Other Items and NULL or UNDEF states are ignored.
+
+### MembersSum
+Get the __SUM__ of direct members' number states.
+```javascript
+var sum = MembersSum(group)
+```
+
+### MembersAvg
+Get the __AVERAGE__ of direct members' number states.
+```javascript
+var avg = MembersAvg(group)
+```
+
+### MembersMin
+Get the __MINIMUM__ of direct members' number states.
+```javascript
+var min = MembersMin(group)
+```
+
+### MembersMax
+Get the __MAXIMUM__ of direct members' number states.
+```javascript
+var max = MembersMax(group)
+```
+
+### allMembersSum
+Get the __SUM__ of all (also child) members' number states.
+```javascript
+var sum = allMembersSum(group)
+```
+
+### allMembersAvg
+Get the __AVERAGE__ of all (also child) members' number states.
+```javascript
+var avg = allMembersAvg(group)
+```
+
+### allMembersMin
+Get the __MINIMUM__ of all (also child) members' number states.
+```javascript
+var min = allMembersMin(group)
+```
+
+### allMembersMax
+Get the __MAXIMUM__ of all (also child) members' number states.
+```javascript
+var max = allMembersMax(group)
+```
+
+### __Count operations__
+
+### countMembers
+Count the direct members matching a given comparison function. 
+Comparison is done through a user-passed function.
+The function returns how many members match the given comparison function.
+```javascript
+var counter = countMembers(group, compareFunc)
+// with the following example scheme:
+var counter = countMembers(group, function (i) { return i.getState() /* comparison operators */ })
```

Comment:
It's a little pedantic, but the `return i.getState()` should have a `;`. It'll work either way but our examples should show best practices where possible.